### ISSUE TITLE
Error when run Node server registry sample demo

### ISSUE THREAD

**Issue by Sneezry** [Azure/azure-iot-sdks#921]
I follow the step in https://github.com/Azure/azure-iot-sdks/blob/master/doc/get_started/node-run-service-sample.md, and I got this error message:

    ErrorCode:InvalidProtocolVersion;BadRequest

Then I changed the version of azure-iothub from `2.0.0-dtpreview` to `1.0.18`.

When I run again, I got this error message:

    Cannot read property 'primaryKey' of undefined at /Users/Sneezry/azure-iot-sdks/node/service/samples/registry_sample.js:20:73

Then I changed `symmetricKey` to `SymmetricKey`, everything is OK.

Please fix this issue. If even our demo can not run correctly, the user may give up our service.

**Comment by pierreca** [comment]
tl;dr: The fix for this is already in the `develop` branch and will be part of the next release. 

What probably happened (just guessing here) is that you tried the SDK for the Device Twin feature preview on a hub that didn't have that preview activated (You can choose to enable that in the portal). if you don't you get the the `InvalidProtocolVersion` error).

The current stable version has had a long-standing bug (the casing of the `SymmetricKey` property) that was corrected in the 2.0.0-dtpreview version (hence the bump to the major version number, because it's a breaking change). 

Because we got some feedback about that already, we will end up shipping the device twin feature in a version 1.1.0 (non-breaking) that will accept both `SymmetricKey` and `symmetricKey`, but accessing keys using the `SymmetricKey` property will be marked as deprecated.

Thanks for the feedback!
